src/qibo/hamiltonians.py

```diff
@@ -186,3 +188,129 @@ def MaxCut(nqubits, dense=True):
     if dense:
         return ham.dense
     return ham
+
+
+from qibo.symbols import X as tspX
+from qibo.symbols import Y as tspY
+from qibo.symbols import Z as tspZ
+class TSP:
+    """
+    This is a TSP class that enables us to implement TSP according to
+    https://arxiv.org/pdf/1709.03489.pdf by Hadfield (2017).
+    Here is an example of how the code can be run.
+
+    num_cities = 3
+    distance_matrix = np.random.rand(num_cities, num_cities)
+    distance_matrix = distance_matrix.round(1)
+    print(distance_matrix)
+    small_tsp = TSP(distance_matrix)
+    obj_hamil, mixer = small_tsp.TspHamiltonians(dense=False)
+    initial_parameters = np.random.uniform(0, 1, 2)
+    initial_state = small_tsp.PrepareInitialStateTsp([i for i in range(num_cities)])
+    qaoa = QAOA(obj_hamil, mixer=mixer)
+
+    """
+    def __init__(self, distance_matrix):
+        """
+        Args:
+            distance_matrix: a numpy matrix encoding the distance matrix.
+        """
+        self.distance_matrix = distance_matrix
+        self.two_to_one = dict()
+        self.num_cities = distance_matrix.shape[0]
+        counter = 0
+        for i in range(self.num_cities):
+            for j in range(self.num_cities):
+                self.two_to_one[(i, j)] = counter
+                counter += 1
+
+
+    def PrepareObjTsp(self, dense=True):
```

Comment:
Not really important, but we tend to use lowercase and _ for method names, eg. these would be `prepare_obj_tsp`. Also I am not sure if it is necessary to include `tsp` in all methods, since they are part of the TSP class anyway.